server/src/main/java/org/apache/druid/segment/metadata/AbstractSegmentMetadataCache.java

```diff
@@ -96,41 +89,37 @@
 import java.util.stream.StreamSupport;
 
 /**
- * Broker-side cache of segment metadata which combines segments to identify
- * datasources which become "tables" in Calcite. This cache provides the "physical"
- * metadata about a datasource which is blended with catalog "logical" metadata
- * to provide the final user-view of each datasource.
+ * An abstract class that listens for segment change events and caches segment metadata and periodically refreshes
```

Comment:
```suggestion
 * An abstract class that listens for segment change events and caches segment metadata. It also periodically refreshes the segments and datasources by xxxx
```

Something like that will help the reader understand the details of this class.